docs/metrics/llm-prompt-metric.md

```diff
@@ -0,0 +1,174 @@
+```
+
+```py
+response = client.chat.completions.create(
+    model="gpt-4-1106-preview",
+    messages=[
+        {
+        "role": "user",
+        "content": INITIAL_PROMPT
+        },
+        {
+        "role": "assistant",
+        "content": "Certainly! Please provide me with the pair of texts for evaluation."
+        },
+        {
+        "role": "user",
+        "content": INPUT_PROMPT
+        },
+    ],
+    temperature=0.5,
+    max_tokens=256,
+    top_p=1
+)
+DECISION = str(response.choices[0].message.content)
+```
+
+### Phase 2 ~ Score Style Prompt
+
+Using GPT-4's output in Phase 1 we query the LLM again and ask for a hallucination score if the LLM has
+previously determined that one exists. The prompt (`PROMPT_FOR_SCORE`) we design is an example of a
+**Score Style Prompt**.
+
+```
+Consider that some hallucinations might hold more significant repercussions in real world applications.
+Provide a numeric hallucination score for the above from 0 to 1.
+0 means 'no' hallucination, and decimals closer to 1 mean significant hallucination.
+Then, on a new line, explain the rating in 10 words.
+```
+
+```py
+response2 = client.chat.completions.create(
+    model="gpt-4-1106-preview",
+    messages=[
+      {
+        "role": "user",
+        "content": INITIAL_PROMPT
+      },
+      {
+        "role": "assistant",
+        "content": "Certainly! Please provide me with the pair of texts for evaluation."
+      },
+      {
+        "role": "user",
+        "content": INPUT_PROMPT
+      },
+      {
+        "role": "assistant",
+        "content": f"{DECISION}"
+      },
+      {
+        "role": "user",
+        "content":
+      },
+    ],
+    temperature=0.5,
+    max_tokens=256,
+    top_p=1
+)
+```
+
+After some post-processing, the output will be something similar to below:
+```py
+(
+    'Yes', # Are the pair of texts hallucinating?
+    'The second text contradicts the first about digestibility.', # Explanation for why
+    '0.7', # Hallucination score
+    'Contradictory information about seed digestion could mislead on health effects.' # Explanation of score
+)
+```
+
+### Chain-of-Thought (CoT) Prompt
+
+Prompt engineering with CoT is similar to the aforementioned two phase approach with GPT-4, however,
+our initial prompt includes a zero-shot chain-of-thought technique that works simply by adding in a
+phrase like "Let's think step by step."
+
+An example of such a prompt would be:
+
+```
+In the context of NLP, a 'hallucination' refers to a phenomenon where the LLM generates text that is incorrect, nonsensical, or not real.
+Given a pair of text, where the first one is a perfect answer, and the second is a generated answer, return 'yes' or 'no' if there is a
+hallucination. If yes, on the next line, explain in about ten words why there is a hallucination. Otherwise, simply return 'no'.
+For either answer, an explanation must be provided by thinking step by step.
```

Comment:
I'm using the prompts that Mark had in his code because he had already pre-computed outputs that I use as the example output. I wasn't sure if changing the prompts would drastically change the output so I left the prompt as is.

Reply:
However, I think the logic is that the model will output only yes or no. And if it gives yes then it will also provide an explanation.